### PR TITLE
Bump the depended Jackson version to 2.15.3

### PR DESCRIPTION
**Pull request by dmikurube**
A fix in Jackson 2.15.3 is needed to enable this fix https://github.com/embulk/embulk-util-json/pull/33#issuecomment-1727651903

https://github.com/FasterXML/jackson-core/pull/1111

It's time to upgrade the depended Jackson to the latest!

### CODE REVIEW

File: src/main/java/org/embulk/util/config/ConfigMapperFactory.java [hunk 1]
```diff
@@ -320,6 +323,39 @@ private ObjectMapper mapperForOthers() {
...
+    }
+
+    private static void assertJacksonDataBindVersion() {
+        if (com.fasterxml.jackson.databind.cfg.PackageVersion.VERSION.getMajorVersion() != 2) {
+            throw new UnsupportedOperationException("embulk-util-config is not used with Jackson 2.");
+        }
+
+        final int minor = com.fasterxml.jackson.databind.cfg.PackageVersion.VERSION.getMinorVersion();
+        if (minor < 14 || (minor == 15 && com.fasterxml.jackson.databind.cfg.PackageVersion.VERSION.getPatchLevel() <= 2)) {
+            throw new UnsupportedOperationException("embulk-util-config is not used with Jackson 2.15.3 or later.");
```
Comment (hiroyuki-sato) [inline src/main/java/org/embulk/util/config/ConfigMapperFactory.java:344]: What do you think about adding `databind` in the log message?
I'm wondering that line 333 and here is the same error message.

File: src/main/java/org/embulk/util/config/ConfigMapperFactory.java [hunk 2]
```diff
@@ -320,6 +323,39 @@ private ObjectMapper mapperForOthers() {
...
+    }
+
+    private static void assertJacksonDataTypeJdk8Version() {
+        if (com.fasterxml.jackson.datatype.jdk8.PackageVersion.VERSION.getMajorVersion() != 2) {
+            throw new UnsupportedOperationException("embulk-util-config is not used with Jackson 2.");
+        }
+
+        final int minor = com.fasterxml.jackson.datatype.jdk8.PackageVersion.VERSION.getMinorVersion();
+        if (minor < 14 || (minor == 15 && com.fasterxml.jackson.datatype.jdk8.PackageVersion.VERSION.getPatchLevel() <= 2)) {
+            throw new UnsupportedOperationException("embulk-util-config is not used with Jackson 2.15.3 or later.");
```
Comment (hiroyuki-sato) [inline src/main/java/org/embulk/util/config/ConfigMapperFactory.java:355]: ditto

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍 Left minor comments.

File: src/main/java/org/embulk/util/config/ConfigMapperFactory.java (hunk 1 above)
Comment (dmikurube) [reply src/main/java/org/embulk/util/config/ConfigMapperFactory.java]: Thanks! Sounds reasonable. Fixed it like `"embulk-util-config is not used with Jackson (jackson-databind) 2.15.3 or later."`

**Comment by dmikurube** [comment]
Thanks for your comments! Addressed them. Merging it!

File: src/main/java/org/embulk/util/config/ConfigMapperFactory.java [hunk 3]
```diff
@@ -320,6 +323,39 @@ private ObjectMapper mapperForOthers() {
...
+
+        final int minor = com.fasterxml.jackson.core.json.PackageVersion.VERSION.getMinorVersion();
+        if (minor < 14 || (minor == 15 && com.fasterxml.jackson.core.json.PackageVersion.VERSION.getPatchLevel() <= 2)) {
+            throw new UnsupportedOperationException("embulk-util-config is not used with Jackson (jackson-core) 2.15.3 or later.");
+        }
+    }
+
+    private static void assertJacksonDataBindVersion() {
+        if (com.fasterxml.jackson.databind.cfg.PackageVersion.VERSION.getMajorVersion() != 2) {
+            throw new UnsupportedOperationException("embulk-util-config is not used with Jackson 2.");
```
Comment (hiroyuki-sato) [inline src/main/java/org/embulk/util/config/ConfigMapperFactory.java:339]: I've missed point out this part. 

Please create another PR when this change useful.
`embulk-util-config is not used with Jackson(jackson-databind) 2.`

**Review by hiroyuki-sato — COMMENTED** [review]
Sorry, I've missed point out another part. Please check for future development.

File: src/main/java/org/embulk/util/config/ConfigMapperFactory.java (hunk 3 above)
Comment (dmikurube) [reply src/main/java/org/embulk/util/config/ConfigMapperFactory.java:339]: I basically don't think it's needed for this part.

Jackson 1 had different Java package names (`org.codehaus.jackson.*`) and Jackson 3 is going to have yet another Java package names (`tools.jackson.*`). While using Jackson 2 (`com.fasterxml.jackson.*`), the major version should be always 2.

This part is practically just an assertion. When it happens, that's not a simple problem anyway. ;)

Comment (hiroyuki-sato) [reply src/main/java/org/embulk/util/config/ConfigMapperFactory.java:339]: I got it. Thanks! LGTM👍

**Pull request merged**